docs/proposals/multi-node/README.md

```diff
@@ -394,35 +410,77 @@ I.e. every time an etcd container restarts, [the old member (represented by the
-
-This approach will not work in the multi-node scenario because we need the individual member pods to be able to talk to each other to maintain the cluster quorum when backup upload fails but need to cut off only client ingress traffic. 
-
-In such a case, we will need two different services. 
+In the multi-node ETCD cluster scenario, it makes sense to distinguish between and separate the peer and client traffic. 
+This can be done by using two `services`.
 
 - peer
   - To be used for peer communication. This could be a `headless` service. 
 - client
-  - To be used for client communication. This could be a normal `ClusterIP` service like it is in the single-node case. 
+  - To be used for client communication. This could be a normal `ClusterIP` service like it is in the single-node case.
+
+The main advantage of this approach is that it makes it possible (if needed) to allow only peer to peer communication while blocking client communication. Such a thing might be required during some phases of some maintenance tasks (manual or automated).
 
-Also, the etcd main container readinessProbe of the member pods will then have to restrict themselves to just the etcd and the sidecar health and readiness and not consider the latest backup upload health which can be done in a different way as follows. 
+### Cutting off client requests
+
+At present, in the single-node ETCD instances, etcd-druid configures the readinessProbe of the etcd main container to probe the healthz endpoint of the etcd-backup-restore sidecar which considers the status of the latest backup upload in addition to the regular checks about etcd and the side car being up and healthy. This has the effect of setting the etcd main container (and hence the etcd pod) as not ready if the latest backup upload failed. This results in the endpoints controller removing the pod IP address from the endpoints list for the service which eventually cuts off ingress traffic coming into the etcd pod via the etcd client service. The rationale for this is to fail early when the backup upload fails rather than continuing to serve requests while the gap between the last backup and the current data increases which might lead to unacceptably large amount of data loss if disaster strikes. 
```

Comment:
Thanks, I guess you inserted that, picking up from https://github.tools.sap/kubernetes-canary/issues-canary/issues/599#issuecomment-183782, where we discussed that fail-fast isn't optimal?

In the beginning, I liked the fail-fast approach, but I came to the realisation, that it does us more harm than good and that what we actually need is alerting. Also, fail-fast doesn't apply here too well, because nothing depends on backups (except for the restores), so raising a red flag isn't usually preventing further harm - it generates it right away by the cluster going down.

Reply:
Agreed. Your Apollo program argument was convincing 🙂